drawee-span/src/main/java/com/facebook/drawee/span/DraweeSpanStringBuilder.java

```diff
@@ -299,43 +326,96 @@ public DrawableChangedListener(DraweeSpan draweeSpan, boolean enableResizing) {
     public DrawableChangedListener(
         DraweeSpan draweeSpan,
         boolean enableResizing,
-        int fixedHeight) {
+        int fixedWidth,
+        int fixedHeight,
+        ScalingUtils.ScaleType scaleType) {
       Preconditions.checkNotNull(draweeSpan);
       mDraweeSpan = draweeSpan;
       mEnableResizing = enableResizing;
+      mFixedWidth = fixedWidth;
       mFixedHeight = fixedHeight;
+      mScaleType = scaleType;
     }
 
     @Override
     public void onFinalImageSet(
         String id,
         ImageInfo imageInfo,
         Animatable animatable) {
-      if (mEnableResizing &&
-          imageInfo != null &&
-          mDraweeSpan.getDraweeHolder().getTopLevelDrawable() != null) {
+
+      if (mEnableResizing && mFixedHeight != UNSET_SIZE && imageInfo != null) {
         Drawable topLevelDrawable = mDraweeSpan.getDraweeHolder().getTopLevelDrawable();
-        Rect topLevelDrawableBounds = topLevelDrawable.getBounds();
-        if (mFixedHeight != UNSET_SIZE) {
-          float imageWidth = ((float) mFixedHeight / imageInfo.getHeight()) * imageInfo.getWidth();
-          int imageWidthPx = (int) imageWidth;
-          if (topLevelDrawableBounds.width() != imageWidthPx ||
-              topLevelDrawableBounds.height() != mFixedHeight) {
-            topLevelDrawable.setBounds(0, 0, imageWidthPx, mFixedHeight);
-
-            if (mDraweeSpanChangedListener != null) {
-              mDraweeSpanChangedListener.onDraweeSpanChanged(DraweeSpanStringBuilder.this);
-            }
-          }
-        } else if (topLevelDrawableBounds.width() != imageInfo.getWidth() ||
-            topLevelDrawableBounds.height() != imageInfo.getHeight()) {
-          topLevelDrawable.setBounds(0, 0, imageInfo.getWidth(), imageInfo.getHeight());
+
+        final Dimens scaledDimens;
+
+        if (mScaleType == ScalingUtils.ScaleType.CENTER) {
+          scaledDimens = getScaledDimensCenter(topLevelDrawable, imageInfo);
+        } else if (mScaleType == ScalingUtils.ScaleType.FIT_START) {
+          scaledDimens = getScaledDimensFitStart(imageInfo);
+        } else {
+          throw new RuntimeException("invalid scaleType: " + mScaleType);
+        }
+
+        if (scaledDimens != null) {
+          topLevelDrawable.setBounds(0, 0, scaledDimens.width, scaledDimens.height);
 
           if (mDraweeSpanChangedListener != null) {
             mDraweeSpanChangedListener.onDraweeSpanChanged(DraweeSpanStringBuilder.this);
           }
         }
       }
     }
+
+    private Dimens getScaledDimensCenter(final Drawable drawable, final ImageInfo imageInfo) {
```

Comment:
I believe I can funnel this computation through `ScalingUtils` instead of computing it here myself -- that work is underway.